activity/scraper.py

```diff
@@ -0,0 +1,95 @@
+import requests
+import json
+import datetime
+from dateutil import parser, relativedelta
+
+# URL to grab all issues from
+issuesurl = "https://coala.github.io/gh-board/issues.json"  # Ignore KeywordBear
+
+# Count of months/weeks/days to be displayed in graph
+month_count = 12
+week_count = 4
+day_count = 7
+today = datetime.datetime.today()
+
+# Calculate the difference in months ignoring partially complete months.
+def diff_month(d1, d2):
+	return (d1.year - d2.year) * 12 + d1.month - d2.month
+
+# Calculate difference in weeks ignoring partially complete weeks.
+def diff_week(d1, d2):
+	monday1 = (d1 - datetime.timedelta(days=d1.weekday()))
+	monday2 = (d2 - datetime.timedelta(days=d2.weekday()))
+	return (monday1 - monday2).days / 7
+
+# Calculate difference in days(ignores and hours difference(since both are dates))
+def diff_days(d1, d2):
+	return (d1-d2).days
+
+# Get labels for each option
+month_labels = []
+for x in range(month_count-1, -1, -1):
+	month_labels.append((today - relativedelta.relativedelta(months=x)).strftime("%B"))
+
+week_labels = []
+for x in range(week_count-1, -1, -1):
+	day = today - relativedelta.relativedelta(weeks=x)
+	week_monday = (day - datetime.timedelta(days=day.weekday()))
+	week_sunday = (day + datetime.timedelta(days=6-day.weekday()))
+	week_labels.append(week_monday.strftime("%b %d") + " - " + week_sunday.strftime("%b %d"))
+
+day_labels = []
+for x in range(day_count-1, -1, -1):
+	day_labels.append((today - datetime.timedelta(days=x)).strftime("%A"))
+
+# Initialise data dicts
+data = {
+	"year" : {
+		"labels": month_labels,
+		"closed": [0]*month_count,
+		"opened": [0]*month_count,
+	},
+	"month" : {
+		"labels": week_labels,
+		"closed": [0]*week_count,
+		"opened": [0]*week_count,
+	},
+	"week" : {
+		"labels": day_labels,
+		"closed": [0]*day_count,
+		"opened": [0]*day_count,
+	},
+}
+
+content = requests.get(issuesurl)
+parsed_json = content.json()
+
+for issue in parsed_json["issues"]:
+	issue = issue["issue"]
+	# Parse date, while ignoring the timestamp.
+	dt = parser.parse(issue["createdAt"][:10])
+
+	mon = diff_month(today, dt)
+	if mon < month_count:
+		mon = month_count - mon
+		data["year"]["opened"][mon] += 1
+		if issue["state"] == "closed":
+			data["year"]["closed"][mon] += 1
+
+	wk = diff_week(today, dt)
+	if wk < week_count:
+		wk = week_count - wk
+		data["month"]["opened"][wk] += 1
+		if issue["state"] == "closed":
+			data["month"]["closed"][wk] += 1
+
+	dys = diff_days(today, dt)
+	if dys < day_count:
+		dys = day_count - dys
+		data["week"]["opened"][dys] += 1
+		if issue["state"] == "closed":
+			data["week"]["closed"][dys] += 1
+
+print(data)
```

Comment:
hmm

Reply:
Printed for logging purposes currently, the rest of the codebase doesn't appear to use proper logging so I just printed it to console too :P